### PR TITLE
Oracle skills tests

### PR DESCRIPTION
**Pull request by jrriehl**
## Proposed changes

- Add skills tests for the oracle skills
- Some small modifications to the ledger test configuration setup

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [x] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/fetchai/contracts/fet_erc20/contract.py [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                    "gasPrice": ledger_api.api.toWei("50", "gwei"),
+                    "nonce": nonce,
+                }
+            )
+            tx = cls._try_estimate_gas(ledger_api, tx)
+            return tx
+        raise NotImplementedError
+
+    @staticmethod
+    def _try_estimate_gas(ledger_api: LedgerApi, tx: Dict[str, Any]) -> Dict[str, Any]:
```
Comment (DavidMinarsch) [inline packages/fetchai/contracts/fet_erc20/contract.py:117]: we now have `update_with_gas_estimate` on the ledger api

File: packages/fetchai/contracts/oracle_client/contract.py [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                    "gasPrice": ledger_api.api.toWei("50", "gwei"),
+                    "nonce": nonce,
+                }
+            )
+            tx = cls._try_estimate_gas(ledger_api, tx)
+            return tx
+        raise NotImplementedError
+
+    @staticmethod
+    def _try_estimate_gas(ledger_api: LedgerApi, tx: Dict[str, Any]) -> Dict[str, Any]:
```
Comment (DavidMinarsch) [inline packages/fetchai/contracts/oracle_client/contract.py:79]: see above

File: packages/fetchai/contracts/oracle_client/contract.yaml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  contract.py: QmT525EcvFFQoNPEfph1PYCgEw3L5erw4nDxDZLqh5j4xZ
+  contracts/FetchOracleTestClient.sol: QmWpUJ4aBrNreiyuXe6EgfSfE7T7hWz3xHDdT7fFye3WCG
+fingerprint_ignore_patterns:
+- build/*
+class_name: FetchOracleClientContract
+contract_interface_paths:
+  ethereum: build/FetchOracleTestClient.json
+dependencies:
+  vyper:
+    version: <1.0.0,>=0.2.7
```
Comment (DavidMinarsch) [inline packages/fetchai/contracts/oracle_client/contract.yaml:19]: don't see any viper

File: packages/fetchai/contracts/fet_erc20/contract.yaml [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+license: Apache-2.0
+aea_version: '>=0.7.0, <0.8.0'
+fingerprint:
+  __init__.py: QmeJ8x6NiZjVLFbUTgpPPCMzQscqjHNBDExapiezpakksR
+  contract.py: QmV8kL37rdgMMgVMx8GrW2f7JuDEcxxXACpSTzK1Scw9QF
+fingerprint_ignore_patterns:
+- build/*
+class_name: FetERC20
+contract_interface_paths:
+  ethereum: build/FetERC20Mock.json
```
Comment (DavidMinarsch) [inline packages/fetchai/contracts/fet_erc20/contract.yaml:15]: missing

File: packages/fetchai/contracts/oracle_client/contract.yaml [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+aea_version: '>=0.7.0, <0.8.0'
+fingerprint:
+  __init__.py: QmRmf3Q5F6nQMrTEt59suBNw6zfgTVQX68E7CAuKmCnhAM
+  contract.py: QmT525EcvFFQoNPEfph1PYCgEw3L5erw4nDxDZLqh5j4xZ
+  contracts/FetchOracleTestClient.sol: QmWpUJ4aBrNreiyuXe6EgfSfE7T7hWz3xHDdT7fFye3WCG
+fingerprint_ignore_patterns:
+- build/*
+class_name: FetchOracleClientContract
+contract_interface_paths:
+  ethereum: build/FetchOracleTestClient.json
```
Comment (DavidMinarsch) [inline packages/fetchai/contracts/oracle_client/contract.yaml:16]: the path specified here does not match the actual path

File: packages/fetchai/skills/simple_oracle_client/behaviours.py [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+        contract_api_dialogue.terms = strategy.get_query_terms()
+        envelope_context = EnvelopeContext(
+            skill_id=self.context.skill_id, connection_id=LEDGER_API_ADDRESS
+        )
+        self.context.outbox.put_message(
+            message=contract_api_msg, context=envelope_context
+        )
+        self.context.logger.info("requesting query transaction...")
+
+    def _get_balance(self):
```
Comment (DavidMinarsch) [inline packages/fetchai/skills/simple_oracle_client/behaviours.py:205]: docstrings missing

File: tests/test_packages/test_skills_integration/test_fetch_beacon.py [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""This test module contains the integration test for the fetch beacon skill."""
+
+from aea.test_tools.test_cases import AEATestCaseEmpty
+
+
+class TestFetchBeaconSkill(AEATestCaseEmpty):
```
Comment (DavidMinarsch) [inline tests/test_packages/test_skills_integration/test_fetch_beacon.py:25]: Test needs to be appropriately marked

**Review by DavidMinarsch — COMMENTED** [review]
Please add readme.md files to all your packages

File: tests/test_packages/test_skills_integration/test_fetch_beacon.py (hunk 7 above)
Comment (jrriehl) [reply tests/test_packages/test_skills_integration/test_fetch_beacon.py:25]: `@pytest.mark.integration'?

Comment (DavidMinarsch) [reply tests/test_packages/test_skills_integration/test_fetch_beacon.py:25]: correct

**Review by DavidMinarsch — COMMENTED** [review]
Please also add `README.md` files for the contract packages

**Review by DavidMinarsch — APPROVED** [review]
LGTM

**Pull request merged**